jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -0,0 +1,118 @@
+from glue.core.message import EditSubsetMessage, SubsetUpdateMessage
+from glue.core.edit_subset_mode import (AndMode, AndNotMode, OrMode,
+                                        ReplaceMode, XorMode)
+from glue.core.subset import (RoiSubsetState, RangeSubsetState,
+                              OrState, AndState, XorState, InvertState)
+from glue_jupyter.widgets.subset_mode_vuetify import SelectionModeMenu
+from traitlets import List, Unicode, Bool, Dict, observe
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin, SubsetSelect
+
+__all__ = ['SubsetPlugin']
+
+SUBSET_MODES = {
+    'replace': ReplaceMode,
+    'add': OrMode,
+    'and': AndMode,
+    'xor': XorMode,
+    'remove': AndNotMode
+}
+
+
+@tray_registry('g-subset-plugin', label="Subset Tools")
+class SubsetPlugin(TemplateMixin):
+    template_file = __file__, "subset_plugin.vue"
+    select = List([]).tag(sync=True)
+    subset_items = List([]).tag(sync=True)
+    subset_selected = Unicode("Create new").tag(sync=True)
+    mode_selected = Unicode('add').tag(sync=True)
+    show_region_info = Bool(False).tag(sync=True)
+    subset_classname = Unicode('').tag(sync=True)
+    subset_definition = Dict({}).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.components = {
+            'g-subset-mode': SelectionModeMenu(session=self.session)
+        }
+
+        self.session.hub.subscribe(self, EditSubsetMessage,
+                                   handler=self._sync_selected_from_state)
+        self.session.hub.subscribe(self, SubsetUpdateMessage,
+                                   handler=self._get_region_definition)
+
+        self.no_selection_text = "Create new"
+        self.subset_select = SubsetSelect(self,
+                                          'subset_items',
+                                          'subset_selected',
+                                          default_text=self.no_selection_text)
```

Comment:
hmmm... maybe we need a `self.subset_select.default_text` read-only property so we don't need this defined in two places?  (Passing the string directly to `default_text` and then replacing all other instances of `self.no_selection_text` with `self.subset_select.default_text` and creating a readonly property in `BaseSelectPluginComponent` that returns `self._default_text`)

Reply:
I took this as Kyle musing on a potential improvement to the subset selector component, not something that needs to be addressed here in this PR.

Reply:
I think it could go in this PR though if you'd like, but probably not necessary.  I'd be happy to do it if its not clear what needs to be done from my musings/ramblings 😜